libcontainer/rootfs_linux.go

```diff
@@ -107,12 +121,12 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 		if err := os.MkdirAll(dest, 0755); err != nil {
 			return err
 		}
-		return syscall.Mount(m.Source, dest, m.Device, uintptr(m.Flags), "")
+		return mountPropagate(m.Source, dest, m.Device, m.Flags, "", m.PropagationFlags)
```

Comment:
We modify m.Destination as well. So that means we will accept rootfs, dest and mountlabel. I find it litle strange. In the method we will allow overrideing m.Destination and m.Data but not other field. But if you are so particular about this method, I can implement it. Let me know.